visualization/app/codeCharta/ui/codeMap/codeMap.label.service.ts

```diff
@@ -96,51 +112,78 @@ export class CodeMapLabelService implements CameraChangeSubscriber {
 		context.font = `${fontsize}px Helvetica Neue`
 
 		const margin = 20
+		const multiLineContext = message.split("\n")
 
 		// setting canvas width/height before ctx draw, else canvas is empty
 		canvas.width = context.measureText(message).width + margin
-		canvas.height = fontsize + margin
+		canvas.height = margin + fontsize * multiLineContext.length
 
 		// bg
-		context.fillStyle = "rgba(255,255,255,1)"
-		context.strokeStyle = ColorConverter.convertHexToRgba(this.storeService.getState().appSettings.mapColors.angularGreen)
+		context.font = `${fontsize}px Helvetica Neue`
+		context.fillStyle = "rgba(224,224,224,0.85)"
 		context.lineJoin = "round"
 		context.lineCap = "round"
 		context.lineWidth = 5
-		context.fillRect(0, 0, canvas.width, canvas.height)
-		context.strokeRect(0, 0, canvas.width, canvas.height)
+
+		this.drawRectangleWithRoundedCorners(context, 0, 0, canvas.width, canvas.height, this.LABEL_CORNER_RADIUS)
 
 		// after setting the canvas width/height we have to re-set font to apply!?! looks like ctx reset
-		context.font = `${fontsize}px Helvetica Neue`
 		context.fillStyle = "rgba(0,0,0,1)"
 		context.textAlign = "center"
 		context.textBaseline = "middle"
-		context.fillText(message, canvas.width / 2, canvas.height / 2)
+
+		for (const [i, element] of multiLineContext.entries()) {
+			context.fillText(element, canvas.width / 2, (canvas.height * (i + 1)) / (multiLineContext.length + 1))
```

Comment:
We could add a comment, why you do this.

Reply:
line 134-136?